backend/models/user.js

```diff
@@ -109,14 +171,18 @@ User.authenticate =  async function (logger, username, password) {
 			username = user.user;
 		}
 
-		authDB = await DB.getAuthDB();
-		await authDB.authenticate(username, password);
-		authDB.close();
-
 		if (!user)  {
 			user = await User.findByUserName(username);
 		}
 
+		if (hasReachedFailedLoginLimit(user)) {
+			throw ({ resCode: responseCodes.TOO_MANY_LOGIN_ATTEMPTS });
+		}
+
+		authDB = await DB.getAuthDB();
```

Comment:
We should move this below L188. There's no point authenticating the user if it's inactive.

Reply:
@carmenfan This actually changes how this function works. At the moment, login/user tests expect an incorrect password to be returned before an error is thrown about an inactive account.

From a user's point of view, there's a difference, but I'm not sure if it matters - if they get their password wrong and their account is also inactive, it'll tell them their account is inactive instead of letting them know they got their password wrong. We can change it to this, but we'll also have to change the tests to match.

Reply:
yea I don't think it matters, if it's inactive it's inactive. It'll probably help support as well